scim2.yaml

```diff
@@ -1640,6 +1640,339 @@ paths:
+        403:
+          description: Forbidden
+          schema:
+            $ref: '#/definitions/ErrorForbidden'
+        404:
+          description: Role is not found
+          schema:
+            $ref: '#/definitions/ErrorRoleNotAvailable'
+        409:
+          description: Role already exist
+          schema:
+            $ref: '#/definitions/ErroRoleAlreadyAvailable'
+  /v3/Roles/.search:
+    post:
+      tags:
+        - Roles Endpoint
+      summary: Search Roles
+      description: |
+        This API returns roles according to the specified filter, sort and pagination parameters.
+
+        <b>Permission required:</b>
+
+            * /permission/admin/manage/identity/rolemgt/create
+      operationId: getRolesByPostV3
+      consumes:
+        - application/scim+json
+      produces:
+        - application/scim+json
+      parameters:
+        - in: body
+          name: body
+          required: false
+          schema:
+            $ref: '#/definitions/RoleSearchRequestObjectV3'
+      responses:
+        200:
+          description: Valid roles are found
+          schema:
+            $ref: '#/definitions/RoleSearchResponseObjectV3'
+        401:
+          description: Unauthorized
+          schema:
+            $ref: '#/definitions/ErrorUnauthorized'
+        403:
+          description: Forbidden
+          schema:
+            $ref: '#/definitions/ErrorForbidden'
+        404:
+          description: Valid roles are not found
+          schema:
+            $ref: '#/definitions/RoleSearchErrorResponseObject'
+  /v3/Roles/{id}:
+    get:
+      tags:
+        - Roles Endpoint
+      summary: Get Role by ID
+      description: |
+        This API returns the role details of a particular role using its unique ID.
+
+        <b>Permission required:</b>
+
+            * /permission/admin/manage/identity/rolemgt/view
+      operationId: getRolebyIdV3
+      produces:
+        - application/scim+json
+      parameters:
+        - name: id
+          in: path
+          description: Unique ID of the resource type.
+          required: true
+          type: string
+        - name: attributes
+          in: query
+          description: SCIM defined attributes parameter.
+          required: false
+          type: string
+        - name: excludedAttributes
+          in: query
+          description: SCIM defined excludedAttribute parameter.
+          required: false
+          type: string
+      responses:
+        200:
+          description: Valid role is found
+          schema:
+            $ref: '#/definitions/RoleGetResponseObjectV3'
+        401:
+          description: Unauthorized
+          schema:
+            $ref: '#/definitions/ErrorUnauthorized'
+        403:
+          description: Forbidden
+          schema:
+            $ref: '#/definitions/ErrorForbidden'
+        404:
+          description: Valid role is not found
+          schema:
+            $ref: '#/definitions/ErrorNoRoleAvailable'
```

Comment:
500 is a possible error response for this API

Reply:
Fixed with: https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/645/commits/2db19bfd2bf84d38ea1c1116c7dd300d49b264d0